### PR TITLE
Task vectors: evaluate and optimize linear combinations of weak-to-strong checkpoints

### PR DESCRIPTION
**Pull request by ojh31**
image

It takes 30gb to run for strong=gpt2-medium/weak=gpt2-small, dtype=bfloat16, so any suggestions to reduce memory usage would be greatly appreciated.

The strange way that parameters are updated is taken from https://discuss.pytorch.org/t/how-does-one-have-the-parameters-of-a-model-not-be-leafs/70076/9?u=pinocchio Every other way that I painstakingly tested today just messes up the computational graph.

### CODE REVIEW

**Comment by ojh31** [comment]
it seems to scale oddly well, pythia-1.4b has >4x as many parameters but is only using 30% extra memory :confused:

**Comment by ojh31** [comment]
I have now added `sweep_task_vector.py` which runs a grid/random/bayes search over the coefficients using wandb as an alternative to gradient descent
